src/util/key.rs

```diff
@@ -182,6 +182,51 @@ impl PublicKey {
         buf
     }
 
+    /// Serialize the public key into a 65 byte array. If `compressed == true` then
+    /// the last 32 bytes in the array will be `0x00`.
+    ///
+    /// This array is not too useful by itself, but it can be used to sort a
+    /// `[PublicKey]` using `sort_by_cached_key`, `sort_by_key`, or any of the other
+    /// `*_by_key` methods on slice. Pass the method into the sort method directly.
+    /// (ie. `PublicKey::get_sort_key`)
+    ///
+    /// If every `PublicKey` in the slice is `compressed == true` then this will sort
```

Comment:
> As arrays are checked in order until the first non-equal value which returns early.

Oh, yes, forgot about this. :sweat_smile:
Moving around those zeroes may still be a bit costly, not sure. Maybe not enough to maintain another function.

I'm surprised the optimizer doesn't seem to be able to optimize-out the comparison. I though it does and [tried to prevent it](https://play.rust-lang.org/?version=nightly&mode=release&edition=2021&gist=da4f3a503d753a8c1eede283c5949e53) but it seems to be same.

Reply:
I have changed this part of the code. Resolving.